proplet/mqtt.go

```diff
@@ -0,0 +1,185 @@
+package proplet
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/url"
+	"time"
+
+	pkgerrors "github.com/absmach/propeller/pkg/errors"
+	mqtt "github.com/eclipse/paho.mqtt.golang"
+)
+
+// NewMQTTClient initializes a new MQTT client with LWT and liveliness updates.
+func NewMQTTClient(config *Config) (mqtt.Client, error) {
+	if err := validateConfig(config); err != nil {
```

Comment:
How many times are we validating the config? This should only happen once when it is loaded